cdap-api/src/main/java/co/cask/cdap/api/workflow/NodeValueEntry.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.workflow;
+
+/**
+ * Multiple nodes in the Workflow can add the same key to the {@link WorkflowToken}.
+ * This class provides a mapping from node name to the value which was set for the
+ * specific key.
+ */
+public final class NodeValueEntry {
+  private final String nodeName;
+  private final String value;
+
+  public NodeValueEntry(String nodeName, String value) {
+    this.nodeName = nodeName;
+    this.value = value;
+  }
+
+  public String getNodeName() {
+    return nodeName;
+  }
+
+  public String getValue() {
+    return value;
+  }
+
+  @Override
+  public String toString() {
+    StringBuilder sb = new StringBuilder("NodeValueEntry {");
+    sb.append("nodeName=").append(nodeName);
+    sb.append(", value=").append(value);
+    sb.append('}');
+    return sb.toString();
+  }
+
+  @Override
+  public boolean equals(Object o) {
```

Comment:
Not sure how the `Objects` class would be useful here for the class which only contain primitive types.
For example, current `equals` method return `nodeName.equals(that.nodeName) && value.equals(that.value)`. Changing it to `Objects.equals(this.nodeName, that.nodeName) && Objects.equals(this.value, that.value)` would be more verbose.. no?